### PR TITLE
adding GHA to publish and deploy Helm Chart

### PR DESCRIPTION
**Pull request by andrestorresGL**
Adding GHA to publish a new release from master branch triggered by `filecoin-project/lotus` repo.

`release.yml`: GHA that creates a new Helm Chart and publish it as a github page.
Secrets needed

- GIT_USERNAME (String)
- GIT_USEREMAIL (String)
- GITHUB_TOKEN (String) needed to create the new release and publish it as a github page.

`deploy.yml`: triggered by `release.yml`, this Action deploys the Helm Chart in a existing K8 cluster and waits until the helm release is up and running.

Secrets needed.

- AWS_ACCESS_KEY_ID (String)
- AWS_SECRET_ACCESS_KEY (String)
- KUBE_CONFIG_DATA (String) this file is the `.kube/config` file used if you don't want to connect to an EKS cluster.

### CODE REVIEW

**Comment by travisperson** [comment]
The release workflow looks like it's intending to replace this:
https://github.com/filecoin-project/helm-charts/blob/b8ef51d2dfd9bb2629c999e0383a0750e54392e1/.circleci/config.yml#L3-L35

Can you make sure this is feature compatible (missing helm lint)?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config user.name "${{ secrets.GIT_USERNAME }}"
+          git config user.email "${{ secrets.GIT_USEREMAIL }}"
```
Comment (travisperson) [inline .github/workflows/release.yml:21]: Are these provided by github directly in the action context or do these need to be provided through repo configuration? If they are not provided are there alternative provided through github actions that can be used?

**Comment by ognots** [comment]
@andrestorresGL what is there some background on the goals of triggering a helm upgrade for the release `lotus-test` ?
Specifically, some information on who is consuming the data from this release and if it's intended to catch bugs in the helm chart, test lotus releases, etc

File: .github/workflows/release.yml (hunk 1 above)
Comment (andrestorresGL) [reply .github/workflows/release.yml:21]: these variables have to be provided through repo configuration.

**Comment by andrestorresGL** [comment]
@ognots this is a POC so you can configure and change later the K8 cluster target. The main goal is the deployment automation for new releases and allow the team to catch any bug.

**Comment by andrestorresGL** [comment]
@travisperson regarding the release proccess, yes the idea is replace the current, as I told @ognots this is a POC, so you may have to include additional behaviors that aren't present on this PR, I'll include the lint test and remove the current release process to avoid the publish race.

**Comment by ognots** [comment]
>  The main goal is the deployment automation for new releases and allow the team to catch any bug.

Which releases?
Which team?

**Comment by travisperson** [comment]
I'm going to close this, please reopen with additional details if this work is still wanted.

**Pull request closed without merging**